opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -276,7 +277,7 @@ String findOriginalName(String fullpath, String changeset) throws IOException {
                     continue;
                 }
 
-                if (commit.getId().getName().startsWith(changeset)) {
+                if (commit.getId().getName().startsWith(changeset)) { // suspicious? - should be an exact match on revision Id
```

Comment:
This should be probably revisited with this change. The javadoc associated with `changeset` says "could be short" which I don't think should be the case anymore.